src/load/azext_load/data_plane/load_test_run/custom.py

```diff
@@ -0,0 +1,549 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+from azext_load.data_plane.utils.utils import (
+    create_or_update_test_run_body,
+    download_file,
+    get_testrun_data_plane_client,
+)
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from azure.core.exceptions import ResourceNotFoundError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def create_test_run(
+    cmd,
+    load_test_resource,
+    test_run_id,
+    test_id,
+    display_name=None,
+    existing_test_run_id=None,
+    description=None,
+    env=None,
+    secrets=None,
+    certificate=None,
+    resource_group_name=None,
+    wait=False,
+):
+    logger.info("Create test run started")
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_body = None
+    try:
+        test_run_body = client.get_test_run(test_run_id=test_run_id)
+    except ResourceNotFoundError:
+        pass
+    if test_run_body:
+        msg = f"Test run with given test run ID : {test_run_id} already exist."
+        logger.debug(msg)
+        raise InvalidArgumentValueError(msg)
+
+    test_run_body = create_or_update_test_run_body(
+        test_id,
+        display_name=display_name,
+        description=description,
+        env=env,
+        secrets=secrets,
+        certificate=certificate,
+    )
+    logger.debug("Creating test run with following request %s", test_run_body)
+    poller = client.begin_test_run(
+        test_run_id=test_run_id,
+        body=test_run_body,
+        old_test_run_id=existing_test_run_id,
+    )
+    response = poller.polling_method().resource()
+    if wait:
+        response = poller.result()
+    logger.info("Test run created with following response %s", response)
+    return response
```

Comment:
For LRO command, when it's no wait, we will not return the initial response.

Reply:
For few commands such as create test and create testrun, the command works in two parts. In both second part is a LRO so if its no wait we want to return response of first part.
For eg -> create test run will create a test run with given details and subsequently run it. test run execution is an LRO. So if a user creates a test run and does not want to wait for test execution duration (can be in hours) should we not return user the initial response of our API which in this case would be the test run object?